astropy/units/format/generic.py

```diff
@@ -346,6 +345,15 @@ def p_function(p):
             '''
             if p[1] == 'sqrt':
                 p[0] = p[3] ** 0.5
+            elif p[1] == 'mag':
+                from ..function.logarithmic import MagUnit
+                p[0] = MagUnit(p[3])
+            elif p[1] == 'dB':
+                from ..function.logarithmic import DecibelUnit
+                p[0] = DecibelUnit(p[3])
+            elif p[1] == 'dex':
+                from ..function.logarithmic import DexUnit
+                p[0] = DexUnit(p[3])
```

Comment:
How difficult would it be to add support for the functions in the FITS, VOUnit and OGIP standards: `log`, `ln` and `exp` while we're in here?

Also, since the FITS and VOUnit parsers inherit from Generic, we should have tests that:

1) Ensure `mag`, `dB` and `dex` are rejected by the FITS and OGIP parsers
2) That `dB` is accepted by the VOUnit parser